lndclient/lnd_services.go

```diff
@@ -206,15 +224,12 @@ func getClientConn(address string, network string, tlsPath string) (
 	// Create a dial options array.
 	opts := []grpc.DialOption{
 		grpc.WithTransportCredentials(creds),
+
+		// Use a custom dialer, to allow connections to unix sockets,
+		// in-memory listeners etc, and not just TCP addresses.
+		grpc.WithContextDialer(dialer),
 	}
 
-	// We need to use a custom dialer so we can also connect to unix sockets
-	// and not just TCP addresses.
-	opts = append(
-		opts, grpc.WithContextDialer(
-			lncfg.ClientAddressDialer(defaultRPCPort),
-		),
-	)
 	conn, err := grpc.Dial(address, opts...)
```

Comment:
Because we always use a dialer that knows the address to dial, the `address` here can just be `""` and doesn't need to be passed through two functions as arguments.

Reply:
Discussed offline, still need address.